soc/riscv32/openisa_rv32m1/linker.ld

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (c) 2013-2014 Wind River Systems, Inc.
+ * Copyright (c) 2016-2017 Jean-Paul Etienne <fractalclone@gmail.com>
+ * Copyright (c) 2018 Foundries.io Ltd
+ *
+ * This file is based on:
+ *
+ * - include/arch/arm/cortex_m/scripts/linker.ld
+ * - include/arch/riscv32/common/linker.ld
+ * - include/arch/riscv32/pulpino/linker.ld
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define _LINKER
+#define _ASMLANGUAGE
+
+#include <autoconf.h>
+
+#include <linker/sections.h>
+#include <linker/linker-defs.h>
+#include <linker/linker-tool.h>
+
+/*
+ * Extra efforts would need to be taken to ensure the IRQ handlers are within
+ * jumping distance of the vector table in non-XIP builds, so avoid them.
+ */
+#ifndef CONFIG_XIP
+#error "non-XIP builds are not supported"
+#endif
+
+#ifdef CONFIG_USERSPACE
+#error "Userspace is not supported"
+#endif
+
+#ifdef CONFIG_BOOTLOADER_MCUBOOT
+#error "Flash partitions (and DTS in general) are not supported"
+#endif
+
+#if defined(CONFIG_CUSTOM_RODATA_LD) || defined(CONFIG_CUSTOM_RWDATA_LD)
+#error "Custom linker data not supported"
+#endif
```

Comment:
It is OK to have these assertions here. But they should redundantly exist in Kconfig as well.